pkg/network/node/networkpolicy.go

```diff
@@ -657,32 +684,24 @@ func (np *networkPolicyPlugin) handleAddOrUpdatePod(obj, _ interface{}, eventTyp
 	// We don't want to grab np.Lock for every Pod.Status change...
 	// But it's safe to look up oldPod without locking here because no other
 	// threads modify this map.
-	oldPod, podExisted := np.pods[pod.UID]
-	if podExisted && oldPod.Status.PodIP == pod.Status.PodIP && reflect.DeepEqual(oldPod.Labels, pod.Labels) {
+	if oldPod != nil && oldPod.Status.PodIP == pod.Status.PodIP && reflect.DeepEqual(oldPod.Labels, pod.Labels) {
 		return
 	}
 
 	np.lock.Lock()
 	defer np.lock.Unlock()
 
-	np.pods[pod.UID] = *pod
-	np.refreshNetworkPolicies(refreshForPods)
+	np.refreshPodNetworkPolicies(pod)
 }
 
 func (np *networkPolicyPlugin) handleDeletePod(obj interface{}) {
 	pod := obj.(*corev1.Pod)
 	klog.V(5).Infof("Watch %s event for Pod %q", watch.Deleted, getPodFullName(pod))
 
-	_, podExisted := np.pods[pod.UID]
-	if !podExisted {
-		return
-	}
-
 	np.lock.Lock()
 	defer np.lock.Unlock()
 
-	delete(np.pods, pod.UID)
-	np.refreshNetworkPolicies(refreshForPods)
+	np.refreshPodNetworkPolicies(pod)
```

Comment:
Does this need to be synchronous?

Reply:
do you have any more specific concern than that?

Note that this is necessarily completely asynchronous with respect to CNI pod creation/deletion anyway

Reply:
Right, but it blocks the pod informer from processing deltas.

Reply:
oh, I see, refreshPodNetworkPolicies is already asynchronous - didn't quite grok it.